packages/twenty-server/src/workspace/calendar-and-messaging/utils/valueStringForBatchRawQuery.util.ts

```diff
@@ -0,0 +1,21 @@
+export const valuesStringForBatchRawQuery = (
+  values: {
+    [key: string]: any;
+  }[],
+  numberOfColumns: number,
+  typesArray?: string[],
+): string => {
+  return values
+    .map((_, row) => {
+      let value = '';
+
+      for (let index = 0; index < numberOfColumns; index++) {
+        value += `$${row * numberOfColumns + index + 1}${
+          typesArray ? `::${typesArray[index]}` : ''
+        }, `;
+      }
+
+      return `(${value.slice(0, -2)})`;
+    })
+    .join(', ');
+};
```

Comment:
I find this more readable that way:

```suggestion
export const valuesStringForBatchRawQuery = (
  values: {
    [key: string]: any;
  }[],
  numberOfColumns: number,
  typesArray: string[] = [],
) => {
  const castedValues = values.reduce((acc, _, rowIndex) => {
    const rowValues = Array.from(
      { length: numberOfColumns },
      (_, columnIndex) => {
        const placeholder = `$${rowIndex * numberOfColumns + columnIndex + 1}`;
        const typeCast = typesArray[columnIndex]
          ? `::${typesArray[columnIndex]}`
          : '';

        return `${placeholder}${typeCast}`;
      },
    ).join(', ');

    acc.push(`(${rowValues})`);

    return acc;
  }, [] as string[]);

  return castedValues.join(', ');
};
```